lib/resources/resource.go

```diff
@@ -0,0 +1,152 @@
+package resources
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	yaml "gopkg.in/yaml.v2"
+
+	"github.com/kosyfrances/rundeck-zabbix/lib/zabbix"
+)
+
+// Zabbix struct holds API info and methods
+type Zabbix struct {
+	API zabbix.API
+}
+
+// Resource struct holds Rundeck resource document mapping info
+type Resource struct {
+	Name        string `json:"host" yaml:"hostname"`
+	Node        string `json:"name" yaml:"nodename"`
+	Description string `json:"description" yaml:"description"`
+	OSArch      string `yaml:"osArch"`
+	OSFamily    string `yaml:"osFamily"`
+	OSName      string `yaml:"osName"`
+	OSVersion   string `yaml:"osVersion"`
+	Tags        string `yaml:"tags"`
+	Username    string `yaml:"username"`
+	SSHKeypath  string `yaml:"ssh-keypath"`
+}
+
+// Result struct holds Zabbix hosts info
+type Result []struct {
+	HostID      string `json:"hostid"`
+	Host        string `json:"host"`
+	Name        string `json:"name"`
+	Description string `json:"description"`
+	Triggers    []struct {
+		Triggerid   string `json:"triggerid,omitempty"`
+		Description string `json:"description,omitempty"`
+		Status      string `json:"status"`
+	} `json:"triggers"`
+}
+
+func (z *Zabbix) getInfo(apiKey, URL string) (Result, error) {
+	api := z.API
```

Comment:
```suggestion
	z.API = zabbix.API{Key: apiKey, URL: URL}
```

Hmm ... also feels a little late in the lifetime of `z` for the contained api to be created 🤔 Can getInfo be called multiple times on the one instance of `z` with varying args: `apiKey`, `URL`? If yes it could be an indication that z has no business holding on to an instance of `API`